src/NUnitFramework/tests/Constraints/CollectionOrderedConstraintTests.cs

```diff
@@ -227,6 +227,46 @@ public void IsOrderedBy_Handles_heterogeneous_classes_as_long_as_the_property_is
             Assert.That(al, Is.Ordered.By("Value"));
         }
 
+        [Test]
+        public void IsOrderedByThenBy()
+        {
+            var al = new List<OrderedByThenByTestClass>();
+            al.Add(new OrderedByThenByTestClass("ABC", 1));
+            al.Add(new OrderedByThenByTestClass("DEF", 2));
+
+            Assert.That(al, Is.Ordered.By("A").Then.By("B"));
+        }
+
+        [Test]
+        public void IsOrderedByDescendingThenBy()
+        {
+            var al = new List<OrderedByThenByTestClass>();
+            al.Add(new OrderedByThenByTestClass("DEF", 1));
+            al.Add(new OrderedByThenByTestClass("ABC", 2));
+
+            Assert.That(al, Is.Ordered.By("A").Descending.Then.By("B"));
+        }
+
+        [Test]
+        public void IsOrderedByThenByDescending()
+        {
+            var al = new List<OrderedByThenByTestClass>();
+            al.Add(new OrderedByThenByTestClass("ABC", 2));
+            al.Add(new OrderedByThenByTestClass("DEF", 1));
+
+            Assert.That(al, Is.Ordered.By("A").Then.By("B").Descending);
+        }
+
+        [Test]
+        public void IsOrderedByDescendingThenByDescending()
+        {
+            var al = new List<OrderedByThenByTestClass>();
+            al.Add(new OrderedByThenByTestClass("DEF", 2));
+            al.Add(new OrderedByThenByTestClass("ABC", 1));
+
+            Assert.That(al, Is.Ordered.By("A").Descending.Then.By("B").Descending);
+        }
+
```

Comment:
Need cases for...
 * Ordered.Descending.By
 * Ordered.By.Descending
 * Ordered.Descending.By.Then.Descending.By
   etc.